packages/gensx-openai/src/structured-output.tsx

```diff
@@ -29,70 +32,153 @@ type StructuredOutputProps<O = unknown> = Omit<
 
 type StructuredOutputOutput<T> = T;
 
+// Helper function to create an output schema tool
+const createOutputSchemaTool = <T,>(
+  outputSchema: z.ZodSchema<T>,
+): GSXTool<any> => {
+  // Create a wrapper schema that will contain the output
+  const wrapperSchema = z.object({
+    output: outputSchema,
+  });
+
+  return GSXTool.create({
+    name: "output_schema",
+    description:
+      "When producing a final output, always call this tool to produce a structured output",
+    schema: wrapperSchema,
+    run: async (args: z.infer<typeof wrapperSchema>) => {
+      // This is a passthrough function - no await needed
+      return Promise.resolve(args);
+    },
+  });
+};
+
 // Extracted implementation function
 export const structuredOutputImpl = async <T,>(
   props: StructuredOutputProps<T>,
 ): Promise<StructuredOutputOutput<T>> => {
-  const { outputSchema, tools: toolsParams, retry, messages, ...rest } = props;
+  const {
+    outputSchema,
+    tools: toolsParams,
+    retry,
+    messages,
+    structuredOutputStrategy = "default",
+    ...rest
+  } = props;
   const maxAttempts = retry?.maxAttempts ?? 3;
   let lastError: Error | undefined;
   let lastResponse: string | undefined;
 
-  const tools = toolsParams?.map((t) =>
-    t instanceof GSXTool ? t : new GSXTool(t),
-  );
+  // Get the OpenAI client from context
+  const context = gensx.useContext(OpenAIContext);
+  if (!context.client) {
+    throw new Error(
+      "OpenAI client not found in context. Please wrap your component with OpenAIProvider.",
+    );
+  }
```

Comment:
@danfhernandez fixed it so we don't need this anymore.

```suggestion
```

Reply:
removing this, thanks!